pkg/endpointmanager/manager.go

```diff
@@ -345,25 +398,52 @@ func RegenerateAllEndpoints(owner endpoint.Owner, regenMetadata *endpoint.Extern
 
 	log.Infof("regenerating all endpoints due to %s", regenMetadata.Reason)
 	for _, ep := range eps {
-		go func(ep *endpoint.Endpoint, wg *sync.WaitGroup) {
-			if err := ep.LockAlive(); err != nil {
-				log.WithError(err).Warnf("Endpoint disappeared while queued to be regenerated: %s", regenMetadata.Reason)
-				ep.LogStatus(endpoint.Policy, endpoint.Failure, "Error while handling policy updates for endpoint: "+err.Error())
-			} else {
-				regen := ep.SetStateLocked(endpoint.StateWaitingToRegenerate, fmt.Sprintf("Triggering endpoint regeneration due to %s", regenMetadata.Reason))
-				ep.Unlock()
-				if regen {
-					// Regenerate logs status according to the build success/failure
-					<-ep.Regenerate(owner, regenMetadata)
-				}
-			}
-			wg.Done()
-		}(ep, &wg)
+		go regenerateEndpoint(owner, ep, regenMetadata, &wg)
+	}
+
+	return &wg
+}
+
+func regenerateEndpoint(owner endpoint.Owner, ep *endpoint.Endpoint, regenMetadata *endpoint.ExternalRegenerationMetadata, wg *sync.WaitGroup) {
```

Comment:
looks out of scope refactoring

Reply:
This functionality is used by `RegenerateEndpointSet` and `RegenerateAllEndpoints`, so factoring it out into a separate function seemed prudent.

Reply:
ah it looked like it was only used in `RegenerateAllEndpoints`